sql/src/main/antlr/OpenSearchSQLLexer.g4

```diff
@@ -438,7 +438,7 @@ BACKTICK_QUOTE_ID:                  BQUOTA_STRING;
 
 // Fragments for Literal primitives
 fragment EXPONENT_NUM_PART:         'E' [-+]? DEC_DIGIT+;
-fragment ID_LITERAL:                [@*A-Z]+?[*A-Z_\-0-9]*;
+fragment ID_LITERAL:                [A-Z]+?[A-Z_\-0-9]*;
```

Comment:
I thought we would have another fragment definition for index names?  We want to preserve the current language for table names

Reply:
Added a separate fragment definition